test/Tests/src/Data/Regression_Spec.enso

```diff
@@ -0,0 +1,105 @@
+from Standard.Base import Nothing, Vector, Number, Decimal, True, Illegal_Argument_Error, False, Regression
+
+import Standard.Test
+
+spec =
+    double_error = 0.000001
+
+    vector_compare values expected =
+        values.each_with_index i->v->
+            case v of
+                Decimal -> v.should_equal (expected.at i) epsilon=double_error
+                _ -> v.should_equal (expected.at i)
+
+    Test.group "Regression" <|
+        Test.specify "return an error if the vector lengths do not match" <|
+            known_xs = [2, 3, 5, 7, 9]
+            known_ys = [4, 5, 7, 10]
+            Regression.fit_least_squares known_xs known_ys . should_fail_with Illegal_Argument_Error
+
+        Test.specify "return an error if the X values are all the same" <|
+            known_xs = [2, 2, 2, 2]
+            known_ys = [4, 5, 7, 10]
+            Regression.fit_least_squares known_xs known_ys . should_fail_with Regression.Fit_Error
+
+        Test.specify "compute the linear trend line" <|
+            known_xs = [2, 3, 5, 7, 9]
+            known_ys = [4, 5, 7, 10, 15]
+            fitted = Regression.fit_least_squares known_xs known_ys
+            fitted.slope . should_equal 1.518292683 epsilon=double_error
```

Comment:
I'm wondering if it's worth linking to this PR as the source for the relevant Excel reference file? Or at least some brief comments how the reference values were computed.